From 38fa1191049ac0c626a6684eea52068dfbbb5078 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michal=20=C4=8Ciha=C5=99?= <michal@cihar.com>
Date: Thu, 25 Feb 2016 09:57:29 +0100
Subject: [PATCH] Urlencode hostname
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This can come from the HTTP header, so we need to be sure to sanitize
it.

Signed-off-by: Michal Čihař <michal@cihar.com>
---
 libraries/Config.class.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libraries/Config.class.php b/libraries/Config.class.php
index 58196732496..4a134042e8c 100644
--- a/libraries/Config.class.php
+++ b/libraries/Config.class.php
@@ -1370,7 +1370,7 @@ public function checkPmaAbsoluteUri()
                 $pma_absolute_uri .= '@';
             }
             // Add hostname
-            $pma_absolute_uri .= $url['host'];
+            $pma_absolute_uri .= urlencode($url['host']);
             // Add port, if it not the default one
             // (or 80 for https which is most likely a bug)
             if (! empty($url['port'])
